gemini/prompts/prompt_optimizer/vertex_ai_prompt_optimizer_sdk_long_prompt_optimization.ipynb

```diff
@@ -0,0 +1,1153 @@
+        "    language=\"English\",  # Language of the system instructions. String. Supported languages: \"English\", \"French\", \"German\", \"Hebrew\", \"Hindi\", \"Japanese\", \"Korean\", \"Portuguese\", \"Simplified Chinese\", \"Spanish\", \"Traditional Chinese\"\n",
+        "    placeholder_to_content=json.loads(\n",
+        "        PLACEHOLDER_TO_CONTENT\n",
+        "    ),  # Placeholder to replace any parameter in the system instruction. Dict.\n",
+        "    data_limit=10,  # Amount of data used for validation. Integer between 5 and 100.\n",
+        ")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "Jd_uzQYQx6L7"
+      },
+      "source": [
+        "#### Upload Vertex AI prompt optimizer Cloud Storage\n",
+        "\n",
+        "After you define Vertex AI prompt optimizer configuration, you upload them on Cloud Storage bucket.\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "iqiv8ApR_SAM"
+      },
+      "outputs": [],
+      "source": [
+        "args = vars(args)\n",
+        "\n",
+        "with epath.Path(CONFIG_FILE_URI).open(\"w\") as config_file:\n",
+        "    json.dump(args, config_file)\n",
+        "config_file.close()"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "spqgBT8hYAle"
+      },
+      "source": [
+        "#### Run the automatic prompt optimization job\n",
+        "\n",
+        "Now you are ready to run your first Vertex AI prompt optimizer job using the Vertex AI SDK for Python."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "RlNQ_UrFH9ne"
+      },
+      "source": [
+        "> This prompt optimization job requires ~ 40 minutes to run.\n",
+        "\n",
+        "> Be sure you have provisioned enough queries per minute (QPM) quota implementing the recommended QPM for each model. If you configure the Vertex AI prompt optimizer with a QPM that is higher than the QPM than you have access to, the job might fail. [Check out](https://cloud.google.com/vertex-ai/generative-ai/docs/learn/prompts/prompt-optimizer#before-you-begin) the documentation to know more.\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "GtPnvKIpUQ3q"
+      },
+      "outputs": [],
+      "source": [
+        "WORKER_POOL_SPECS = [\n",
+        "    {\n",
+        "        \"machine_spec\": {\n",
+        "            \"machine_type\": \"n1-standard-4\",\n",
+        "        },\n",
+        "        \"replica_count\": 1,\n",
+        "        \"container_spec\": {\n",
+        "            \"image_uri\": APD_CONTAINER_URI,\n",
+        "            \"args\": [\"--config=\" + CONFIG_FILE_URI],\n",
+        "        },\n",
+        "    }\n",
+        "]\n",
```

Comment:
Is there a built-in class for this?

Reply:
Yes, but I aligned with the official documentation.